### DISCUSSION TITLE
Save relationship MorphTo

### DISCUSSION THREAD

**Discussion by raorsa** [Laravel-Backpack/community-forum#823] [Show and Tell]
I have a MorphTo relationship that is not saved automatically

        CRUD::field([
            'name' => 'operationElement',
            'label' => "Elemento de operacion",
        ])->addMorphOption('App\Models\Intervention', 'Registro de intervencion')
            ->addMorphOption('App\Models\Repair', 'Reparaciones electronicas');

But

Create.php only have this options in switch

                case 'HasOne':
                case 'MorphOne':

                case 'HasMany':
                case 'MorphMany':

                case 'BelongsToMany':
                case 'MorphToMany':

**Comment by jcastroa87** [comment]
Hello @raorsa 
Can you share your model please.

Did you try follow this [docs](https://backpackforlaravel.com/docs/6.x/crud-how-to#morphto-n-1-relationship)?

Cheers.

**Comment by raorsa** [comment]
class Activity extends Model

    public function operationElement(): MorphTo
    {
        return $this->morphTo(__FUNCTION__, 'cri_type', 'cri_id');
    }


class Intervention extends Model implements \Stringable
    public function activities(): MorphMany
    {
        return $this->morphMany(Activity::class, 'operationElement', 'cri_type', 'cri_id');
    }


class Repair extends Model implements \Stringable
    public function activities(): MorphMany
    {
        return $this->morphMany(Activity::class, 'operationElement', 'cri_type', 'cri_id');
    }

**Comment by jcastroa87** [comment]
Hi @raorsa on my side is working as expected i follow the example on the demo: https://github.com/Laravel-Backpack/demo/blob/main/app/Http/Controllers/Admin/PetShop/CommentCrudController.php and model https://github.com/Laravel-Backpack/demo/blob/main/app/Models/PetShop/Comment.php you can check everything is working with a "morphTo".
Can you try to follow the example to check if the demo is working on your side too? Thanks

Cheers.

**Reply by raorsa** [reply]
This example works because set fillable commentable_type and commentable_id, but i have impression this method it's not very elegant
